test/Microsoft.ML.Tests/OnnxConversionTest.cs

```diff
@@ -2145,6 +2146,171 @@ public void NonDefaultColNamesMultiClassificationOnnxConversionTest()
+        {
+            var chain = new EstimatorChain<ITransformer>().Append(pipeline);
+            TestPipeline(mlContext, chain, dataView, onnxFileName, columnsToCompare);
+        }
+
+        /// <summary>
+        /// Helper method that takes an EstimatorChain{TLastTransformer} and an IEnumerable{TRow} and converts the IEnumerable{TRow} into an IDataView and then
+        /// calls the actual testing method.
+        /// </summary>
+        /// <typeparam name="TLastTransformer">The type of for the EstimatorChain. Will be auto detected.</typeparam>
+        /// <typeparam name="TRow">The type of the IEnumerable. Will be auto detected.</typeparam>
+        /// <param name="mlContext">The MLContext</param>
+        /// <param name="pipeline">A single IEstimator{ITransformer} to be tested.</param>
+        /// <param name="data">The test data as a IEnumerable{TRow}.</param>
+        /// <param name="onnxFileName">Name to save the ONNX model file.</param>
+        /// <param name="columnsToCompare">Which columns you want to compare. This assumes that the column name in ONNX is the same as ML.Net, so only 1 name per column is provided.</param>
+        /// <param name="schemaDefinition">Optional schema definition for the IEnumerable{TRow}.</param>
+        private void TestPipeline<TLastTransformer, TRow>(MLContext mlContext, EstimatorChain<TLastTransformer> pipeline, IEnumerable<TRow> data, string onnxFileName, string[] columnsToCompare, SchemaDefinition schemaDefinition = null)
+            where TLastTransformer : class, ITransformer
+            where TRow : class
+        {
+            var dataView = mlContext.Data.LoadFromEnumerable(data, schemaDefinition);
+            TestPipeline(mlContext, pipeline, dataView, onnxFileName, columnsToCompare);
+        }
+
+        /// <summary>
+        /// Testing method that takes an EstimatorChain{TLastTransformer} and an IDataView and converst the IEnumerable{TRow} into an IDataView and then
+        /// converts the chain to an ONNX model and compares the results.
+        /// </summary>
+        /// <typeparam name="TLastTransformer">The type of for the EstimatorChain. Will be auto detected.</typeparam>
+        /// <param name="mlContext">The MLContext</param>
+        /// <param name="pipeline">A single IEstimator{ITransformer} to be tested.</param>
+        /// <param name="dataView">The test data.</param>
+        /// <param name="onnxFileName">Name to save the ONNX model file.</param>
+        /// <param name="columnsToCompare">Which columns you want to compare. This assumes that the column name in ONNX is the same as ML.Net, so only 1 name per column is provided.</param>
+        private void TestPipeline<TLastTransformer>(MLContext mlContext, EstimatorChain<TLastTransformer> pipeline, IDataView dataView, string onnxFileName, params string[] columnsToCompare)
+            where TLastTransformer : class, ITransformer
```

Comment:
Other tests have used a new MLContext object for their tests but it is strictly not needed. The base test class already has one and that can be reused. 
So it would be good to not have MLContext as a param here and just use the object from the base class.

Reply:
Good point. It is now removed.